### DISCUSSION TITLE
Get timeframe of the chart

### DISCUSSION THREAD

**Discussion by jppeltier** [Mathieu2301/TradingView-API#43] [Q&A] [answered]
Hello Mathieu,

I am browsing through `chart.infos` and there are many useful infos, but ... I seem not to be able to find the timeframe used (daily, 4H ...). I tried `chart.currentSeries` but it remains `undefined`. Did I miss it, or is there another way to get it?

Thanks,

**Comment by Mathieu2301** [comment, marked as answer]
I don't understand why would you like to get the used timeframe since you set it when you call `setMarket` or `setSeries`.
The default timeframe is '240' (4H).

**Reply by jppeltier** [reply]
For controlling purposes, as we are in asynchronous context, I'd rather make sure that the data I am collecting at a given time are on the proper asset and timeframe, if I browse through assets and timeframes.

_Edit_: I confirm what I feared. In the `chart.onUpdate` callback, even if you changed the asset through `chart.setMarket` in the meantime, you still can check to what asset this callback belongs through the `chart.infos`. But if you changed the timeframe, you can't check in the `chart.onUpdate` callback if the data belongs to the previous or the new timeframe. By extension, I fear we will probably have an even bigger issue with indicators, as we cannot know to which asset and timeframe belongs a `study.onUpdate` callback (maybe implement a `chart` reference in the `study` object, pointing back to the parent `chart` object, to be able to access its `chart.infos`?).

P.S.: Merry Christmas! :-)

**Reply by Mathieu2301** [reply]
Hmmm okay I see...
I just checked and I saw that the server never sends the timeframe after a change.
So I think the only way to know it is by using the `onSymbolLoaded` event.

**Reply by Mathieu2301** [reply]
By doing something like this:
```javascript
const client = new TradingView.Client();
const chart = new client.Session.Chart();

let wantedTimeframe = '5';
let timeframe = null;

chart.setMarket('BINANCE:BTCEUR', {
  timeframe: wantedTimeframe,
});

chart.onSymbolLoaded(() => {
  timeframe = wantedTimeframe;
  console.log(`Timeframe is now '${timeframe}'`);
});

chart.onUpdate(() => {
  console.log(`Received data on '${timeframe}' timeframe`);
});

setTimeout(() => {
  wantedTimeframe = '240';
  chart.setMarket('BINANCE:BTCEUR', {
    timeframe: wantedTimeframe,
  });
}, 3000);
```

The result:
```ruby
Timeframe is now '5'
Received data on '5' timeframe
Received data on '5' timeframe
Received data on '5' timeframe
Received data on '5' timeframe
Timeframe is now '240'
Received data on '240' timeframe
Received data on '240' timeframe
Received data on '240' timeframe
...
```
